### DISCUSSION TITLE
Issue using Zip backend

### DISCUSSION THREAD

**Discussion by lvcabral** [zen-fs/core#144] [General]
I tried in two different TypeScript projects to use the Zip backend, but keep getting this issue:
![image](https://github.com/user-attachments/assets/aaa5de4e-5e8e-4f82-8bed-1b578aec714d)

**Comment by lvcabral** [comment]
And I do have the declaration correct
![image](https://github.com/user-attachments/assets/b0dcbc5b-c638-4dcc-b86a-16ff5be85b1a)

**Comment by lvcabral** [comment]
Tried to use `_Zip`  and the VSCode issue disappeared but the build failed:
![image](https://github.com/user-attachments/assets/9a421467-2b5c-4c69-8767-25df7d2cd5f7)

**Comment by james-pre** [comment]
@lvcabral, Please make sure your environment has declarations and support for ES decorators (not legacy decorators) as well as *declarations* for `Symbol.metadata`. You shouldn't need runtime support for `Symbol.metadata` as Utilium will polyfill it.

**Reply by lvcabral** [reply]
Not sure what you mean by that, this is my tsconfig.json
![image](https://github.com/user-attachments/assets/b379ff18-281e-484f-b3de-03b8835bfadf)

**Reply by lvcabral** [reply]
It builds with `core` but not with Zip, what should we have different to support Zip backend?

**Reply by lvcabral** [reply]
I added "ESNExt" to the libs, and the error changed to:

![image](https://github.com/user-attachments/assets/2bd9257d-ab6c-4527-9ba7-5894b46c62e1)

**Reply by lvcabral** [reply]
And should I use `_Zip` ? Why `Zip` does not work?

**Reply by james-pre** [reply]
Please use the exports from `dist`, not `src`- the source imports will not work correctly and will be removed soon.

> And should I use `_Zip`? Why `Zip` does not work?

`Zip` is just an alias for `_Zip`, since using the object directly leads to very nasty error messages.

**Reply by lvcabral** [reply]
Fixed updating my config to this:
![image](https://github.com/user-attachments/assets/2ee83df8-ffe2-45f2-b910-337ee2d34f32)

**Reply by lvcabral** [reply]
Thanks again for the support.

**Reply by james-pre** [reply]
Yeah, of course!
